src/java.base/share/classes/java/security/MessageDigest.java

```diff
@@ -176,30 +177,27 @@ private MessageDigest(String algorithm, Provider p) {
      * @see Provider
      */
     public static MessageDigest getInstance(String algorithm)
-    throws NoSuchAlgorithmException {
+        throws NoSuchAlgorithmException
+    {
         Objects.requireNonNull(algorithm, "null algorithm name");
-        try {
-            MessageDigest md;
-            Object[] objs = Security.getImpl(algorithm, "MessageDigest",
-                                             (String)null);
-            if (objs[0] instanceof MessageDigest) {
-                md = (MessageDigest)objs[0];
-                md.provider = (Provider)objs[1];
-            } else {
-                md = Delegate.of((MessageDigestSpi)objs[0], algorithm,
-                    (Provider) objs[1]);
-            }
-
-            if (!skipDebug && pdebug != null) {
-                pdebug.println("MessageDigest." + algorithm +
-                    " algorithm from: " + md.provider.getName());
-            }
+        MessageDigest md;
 
-            return md;
+        GetInstance.Instance instance = GetInstance.getInstance("MessageDigest",
```

Comment:
Sure, why not. The Security.getImpl has a branch checking null provider which is pointless here since we guard against a null provider before that call, so we can reuse the same pattern here.